src/modules/scene/vscene.ts

```diff
@@ -346,10 +342,17 @@ export class VScene {
             }
             const avatarHeight = boundingVectors.max.y - boundingVectors.min.y;
 
+            const hipPosition = result.skeleton?.bones.find((bone) => bone.name === "Hips")?.position;
```

Comment:
`hipPosition` may be undefined? What happens then?

Reply:
Ah, good catch.

`hipPosition` is currently only used by the `loadAvatarAnimations` method.
If `hipPosition` is undefined (the hip bone can't be found or doesn't exist), then it will default to a `Vector3.Zero()` when passed into `loadAvatarAnimations`:
```typescript
// resource.ts @ 67

public async loadAvatarAnimations(modelUrl: string, hipPosition = Vector3.Zero()): Promise<IAvatarAnimationResult> {
//                                                  ^
```

This results in the avatar's hip bone being treated as if it were at `0,0,0` (on the ground), offsetting the animation model's hip bone by its own height:
```typescript
// resource.ts @ 103

offset = animationInitialHipPosition.y - hipPosition.y; // hipPosition.y will be 0.
```

I guess we have two options here:
1. Use this logic. Offsetting the animations so that they originate from `0,0,0`.
2. Ignore the offset when this value is `0,0,0`:
  ```typescript
  if (hipPosition.y !== 0) {
    offset = animationInitialHipPosition.y - hipPosition.y;
  }
  ```

I'm not sure if we're planning to support avatars that don't have a hip bone. If not, then we should probably switch to the fallback avatar if a hip bone isn't found.

What are your thoughts?
And @Aitolda what are your thoughts?

Reply:
All good points - looks like there's actually a lot to consider here. I'll make this a topic for this Friday's dev meeting.

For now I'll implement option 2, since it won't have a noticeable effect on arbitrary meshes and avatars that don't have a hip bone.

Reply:
Made this adjustment with commit [39d4603](https://github.com/vircadia/vircadia-web/pull/130/commits/39d460309fac1c1d049604b0305a283841779da2). This means that arbitrary meshes and avatars that don't have a hip bone won't see any change from this PR.